pkg/resources/kafka/configmap.go

```diff
@@ -250,3 +255,116 @@ func (r Reconciler) generateBrokerConfig(id int32, brokerConfig *v1beta1.BrokerC
 }
+
+func (r *Reconciler) postConfigMapReconcile(log logr.Logger, desired *corev1.ConfigMap) error {
+	desiredType := reflect.TypeOf(desired)
+	current := desired.DeepCopyObject()
+
+	key, err := runtimeClient.ObjectKeyFromObject(current)
+	if err != nil {
+		return errors.WithDetails(err, "kind", desiredType)
+	}
+	log = log.WithValues("kind", desiredType, "name", key.Name)
+
+	err = r.Client.Get(context.TODO(), key, current)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errorfactory.New(errorfactory.APIFailure{}, err, "getting resource failed", "kind", desiredType, "name", key.Name)
+	}
+
+	if k8sutil.CheckIfObjectUpdated(log, desiredType, current, desired) {
+		// Only update status when configmap belongs to broker
+		if id, ok := desired.Labels["brokerId"]; ok {
+			currentConfigs := getBrokerConfigsFromConfigMap(current.(*corev1.ConfigMap))
+			desiredConfigs := getBrokerConfigsFromConfigMap(desired)
+
+			var statusErr error
+			// if only per broker configs are changed, do not trigger rolling upgrade by setting ConfigOutOfSync status
+			if shouldRefreshOnlyPerBrokerConfigs(currentConfigs, desiredConfigs, log) {
+				log.V(1).Info("setting per broker config status to out of sync")
+				statusErr = k8sutil.UpdateBrokerStatus(r.Client, []string{id}, r.KafkaCluster, v1beta1.PerBrokerConfigOutOfSync, log)
+			} else {
+				statusErr = k8sutil.UpdateBrokerStatus(r.Client, []string{id}, r.KafkaCluster, v1beta1.ConfigOutOfSync, log)
+			}
+			if statusErr != nil {
+				return errors.WrapIfWithDetails(err, "updating status for resource failed", "kind", desiredType)
+			}
+		}
+	}
+
+	return nil
+}
+
+func shouldRefreshOnlyPerBrokerConfigs(currentConfigs, desiredConfigs map[string]string, log logr.Logger) bool {
+	touchedConfigs := collectTouchedConfigs(currentConfigs, desiredConfigs, log)
+
+OUTERLOOP:
+	for _, config := range touchedConfigs {
+		// Security protocol cannot be updated for existing listener
+		// a rolling upgrade should be triggered in this case
+		if config == securityProtocolMap {
+			// added or deleted config is ok
+			if currentConfigs[securityProtocolMap] == "" || desiredConfigs[securityProtocolMap] == "" {
+				continue
+			}
+			currentListenerProtocolMap := make(map[string]string, 0)
+			desiredListenerProtocolMap := make(map[string]string, 0)
+			for _, listenerConfig := range strings.Split(currentConfigs[securityProtocolMap], ",") {
+				listenerKeyValue := strings.Split(listenerConfig, ":")
+				if len(listenerKeyValue) != 2 {
+					continue
+				}
+				currentListenerProtocolMap[listenerKeyValue[0]] = listenerKeyValue[1]
+			}
+			for _, listenerConfig := range strings.Split(desiredConfigs[securityProtocolMap], ",") {
+				listenerKeyValue := strings.Split(listenerConfig, ":")
+				if len(listenerKeyValue) != 2 {
+					continue
+				}
+				desiredListenerProtocolMap[listenerKeyValue[0]] = listenerKeyValue[1]
+			}
+
+			for protocolName, desiredListenerProtocol := range desiredListenerProtocolMap {
```

Comment:
There is no need for the `for` loop. The same verification can be done in the `for` loop above instead of populating `desiredListenerProtocolMap`

```
```